linux/mcompile/keymap/filesystem.cpp

```diff
@@ -0,0 +1,188 @@
+
+#ifdef __EMSCRIPTEN__
+#include <emscripten.h>
+#include <emscripten/bind.h>
+#endif
+
+#include <cassert>
+#include <algorithm>
+#include <iostream>
+#include "filesystem.h"
+
+#ifdef _MSC_VER
+#include <io.h>
+#else
+#include <unistd.h>
+#endif
+
+//#define _DEBUG_FOPEN  1
+
+#ifdef __EMSCRIPTEN__
+
+const std::string get_wasm_file_path(const std::string& filename) {
+  // Verify that we are passing a fully-qualified path
+  // TODO: we need to support relative paths based on CWD
+#if _DEBUG_FOPEN
+  std::cout << "get_wasm_file_path ENTER (" << filename << ")" << std::endl;
```

Comment:
You could use the `__FUNCTION__` macro instead of hard coding the method name. That way it'll output correctly even if the method gets renamed:


```suggestion
  std::cout << __FUNCTION__ << " ENTER (" << filename << ")" << std::endl;
```

Reply:
This is used in mcompile-linux and mcompile-mac. 
In order to have the same code of mcompile throughout all OS´s I moved this to a new issue  [( # 11802)](https://github.com/keymanapp/keyman/issues/11802)